src/main/kotlin/sideChain/iroha/consumer/IrohaConsumerImpl.kt

```diff
@@ -0,0 +1,124 @@
+package sideChain.iroha.consumer
+
+import ByteVector
+import ModelProtoTransaction
+import ModelTransactionBuilder
+import PublicKey
+import iroha.protocol.BlockOuterClass
+import mu.KLogging
+import notary.IrohaCommand
+import notary.IrohaOrderedBatch
+import notary.IrohaTransaction
+import notary.Notary
+import java.math.BigInteger
+import kotlin.reflect.jvm.internal.impl.protobuf.InvalidProtocolBufferException
+
+/**
+ * Endpoint of Iroha to write transactions
+ * @param source that emits [IrohaCommand]s
+ * @param cryptoProvider provides with Iroha keypair
+ * @param irohaNetwork network layer of Iroha
+ */
+class IrohaConsumerImpl(
+    source: io.reactivex.Observable<IrohaOrderedBatch>,
+    cryptoProvider: IrohaCryptoProvider, val irohaNetwork: IrohaNetworkImpl
+) : IrohaConsumer {
+
+    /** Iroha keypair for signing */
+    val keypair = cryptoProvider.keypair
+
+    init {
+        source.subscribe {
+            irohaOutput(it)
+        }
+    }
+
+    /** Convert [ByteVector] to [ByteArray] */
+    private fun toByteArray(blob: ByteVector): ByteArray {
```

Comment:
Move this and further method to own Iroha Util package for improving testability.